controller/proxy-injector/webhook.go

```diff
@@ -129,11 +131,20 @@ func (w *Webhook) inject(request *admissionv1beta1.AdmissionRequest) (*admission
 
 	p.AddCreatedByPodAnnotation(fmt.Sprintf("linkerd/proxy-injector %s", version.Version))
 
-	// When adding workloads through `kubectl apply` the spec template labels are
-	// automatically copied to the workload's main metadata section.
-	// This doesn't happen when adding labels through the webhook. So we manually
-	// add them to remain consistent.
-	conf.AddRootLabels(p)
+	if refs := conf.GetOwnerReferences(); len(refs) > 0 {
+		// assuming just one owner reference
+		k, v, err := w.parentRefLabel(request.Namespace, refs[0])
+		if err != nil {
+			return nil, err
+		}
+		p.AddPodLabel(k, v)
+	} else {
+		// When adding workloads through `kubectl apply` the spec template labels are
+		// automatically copied to the workload's main metadata section.
+		// This doesn't happen when adding labels through the webhook. So we manually
+		// add them to remain consistent.
+		conf.AddRootLabels(p)
```

Comment:
IIUC, if the workload is a single pod, this will just re-add the labels to its `metadata`, right? I *think* issuing `add` patches to re-add existing labels won't break anything, at least, that's the behaviour with `kubectl`.

Reply:
I think the `else` part of that snippet and `AddRootLabels()` don't make sense anymore. That was for updating the `Deployment` metadata, which is no longer there since we're watching Pods. You agree?

Reply:
Yeah, I think removing it makes sense. I tested this further with `kubectl`. If I tried to re-annotate a workload with the same value, it errors out if there isn't an `--overwrite` flag, which leads me to think sending an `add` patch again will trigger the same error.

```
k annotate pod nginx-overrides-5d5b675ddd-cznwh linkerd.io/config=enabled
error: --overwrite is false but found the following declared annotation(s): 'linkerd.io/config' already has a value (enabled)
```